src/domains/dashboard/components/node-view/node-view.tsx

```diff
@@ -208,14 +216,31 @@ export const NodeView = ({
   }, [width])
 
 
+  const { y } = useWindowScroll()
+  useEffect(() => {
+    if (ref.current) {
+      const currentNode = Array.from(ref.current.querySelectorAll("[id]"))
+        .filter((node) => (node.getAttribute("id") as string).startsWith("menu"))
+        .find((node) => node.getBoundingClientRect().top > 0)
```

Comment:
filter could be not used here due to correct function in find. 
For example
```
const menuWithRect => (node: any) =>  {
(node.getAttribute("id") as string).startsWith("menu")) && 
node.getBoundingClientRect().top > 0
}
...........
Array.from(ref.current.querySelectorAll("[id]")).find(menuWithRect)
```

Reply:
good idea to separate that function. Not sure if merging is better than composition of 2, but it's not that complicated, so all good